src/controllers/signAccountOp/signAccountOp.ts

```diff
@@ -351,7 +382,13 @@ export class SignAccountOpController extends EventEmitter {
   }
 
   get readyToSign() {
-    return !!this.status && this.status?.type === SigningStatus.ReadyToSign
+    const criticalErrors = this.errors.filter(
+      (error) => !Object.values(NON_CRITICAL_ERRORS).includes(error)
+    )
+
+    return (
+      !!this.status && this.status?.type === SigningStatus.ReadyToSign && !criticalErrors.length
```

Comment:
why do we have to check for critical errors here? shouldn't the rest of the code already ensure that `this.status` can't be changed to ReadyToSign if we encounter a critical error?

Reply:
agreed